addon/mixins/route.js

```diff
@@ -205,72 +194,98 @@ export default Ember.Mixin.create({
    @return {Boolean}
    */
   _infinityLoad() {
-    var nextPage    = this.get('currentPage') + 1;
-    var perPage     = this.get('_perPage');
-    var totalPages  = this.get('_totalPages');
-    var modelName   = this.get('_infinityModelName');
-    var options     = this.get('_extraParams');
-    var boundParams = this.get('_boundParams');
-
-    if (!this.get('_loadingMore') && this.get('_canLoadMore')) {
-      this.set('_loadingMore', true);
-
-      var requestPayloadBase = {};
-      requestPayloadBase[this.get('perPageParam')] = perPage;
-      requestPayloadBase[this.get('pageParam')] = nextPage;
-
-      options = this._includeBoundParams(options, boundParams);
-      var params = Ember.merge(requestPayloadBase, this.get('_extraParams'));
-
-      let promise = this.store[this._storeFindMethod](modelName, params);
-
-      promise.then(
-        newObjects => {
-
-          this.updateInfinityModel(newObjects);
-          this.set('_loadingMore', false);
-          this.set('currentPage', nextPage);
-          if(this.infinityModelUpdated) {
-            Ember.run.scheduleOnce('afterRender', this, 'infinityModelUpdated', {
-              lastPageLoaded: nextPage,
-              totalPages: totalPages,
-              newObjects: newObjects
-            });
-          }
-          if (!this.get('_canLoadMore')) {
-            this.set(this.get('_modelPath') + '.reachedInfinity', true);
-            if(this.infinityModelLoaded) {
-              Ember.run.scheduleOnce('afterRender', this, 'infinityModelLoaded', {
-                totalPages: totalPages
-              });
-            }
-          }
-        },
-        () => {
-          this.set('_loadingMore', false);
-          throw new Ember.Error("Ember Infinity: Could not fetch Infinity Model. Please check your serverside configuration.");
-        }
-      );
-    } else {
-      if (!this.get('_canLoadMore')) {
-        this.set(this.get('_modelPath') + '.reachedInfinity', true);
-        if(this.infinityModelLoaded) {
-          Ember.run.scheduleOnce('afterRender', this, 'infinityModelLoaded', { totalPages: totalPages });
-        }
-      }
+    if (this.get('_loadingMore') || !this.get('_canLoadMore')) {
+      this._finishLoad();
+      return false;
     }
```

Comment:
the original code branch was:

```javascript
if (!this.get('_loadingMore') && this.get('_canLoadMore')) {
  // load more
  ...
} else {
  ...
  // equivalent of finishLoad()
}
```

Here, I've inverted the condition and flattened:

```javascript
if (this.get('_loadingMore') || !this.get('_canLoadMore')) {
  this._finishLoad();
}

// load more
```

The logic _should_ be equivalent to before, just more easy to see what's going on. If that isn't correct, that may be out of scope.